.circleci/config.yml

```diff
@@ -62,15 +63,49 @@ jobs:
           command: |
             ./.circleci/set-nameserver.sh sudo
             sudo apt-get install gettext-base
-      - attach_workspace:
-          at: ./data
+      - run:
+          command: |
+            sudo mkdir -p /cncf/data
+            sudo chown -R circleci:circleci /cncf/
+          name: Cache Prep
+      - restore_cache:
+          key: cncf-data-{{.Environment.CIRCLE_WORKFLOW_ID}}
+      - run:
+          command: |
+            mv /cncf/data ./data
+            find ./data
+          name: Move Cache
       - run:
           command: ./scripts/circle-integration-tests.sh
           environment:
             KUBECONFIG: /go/src/github.com/ligato/networkservicemesh/data/kubeconfig
       - run:
-          when: always
-          command: ./.circleci/destroy-cluster.sh
+          when: on_fail
+          name: Trigger packet-destroy
+          command: |
+            curl --user ${CIRCLE_API_PROJECT_TOKEN}: \
+                --data build_parameters[CIRCLE_JOB]=packet-destroy \
+                --data build_parameters[CIRCLE_WORKFLOW_ID]=${CIRCLE_WORKFLOW_ID} \
+                --data revision=$CIRCLE_SHA1 \
+                https://circleci.com/api/v1.1/project/github/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/tree/$CIRCLE_BRANCH
+
+
+  packet-destroy:
+    working_directory: /cncf
+    docker:
+      - image: registry.cncf.ci/cncf/cross-cloud/provisioning:production
+    steps:
+      - restore_cache:
+          key: cncf-data-{{.Environment.CIRCLE_WORKFLOW_ID}}
+      - run: find /cncf/data
+      - checkout:
+          path:
+            nsm
+      - run: nsm/.circleci/set-nameserver.sh
+      - run: nsm/.circleci/provision.sh packet-destroy
+      - run:
+          when: on_fail
+          command: nsm/.circleci/provision.sh packet-destroy
```

Comment:
@edwarnicke should we remove on_fail -> packet_destroy here (in packet-destroy job)?

Reply:
@platoff I'm honestly OK with either keeping it or leaving it.  My thinking in leaving it had been that in the event of some kind of problem destroying the cluster, it would 'retry' destroy.  That said, running from the provision container for destroy seems super reliable, so it may not be necessary.

Do you have any strong opinions here?

Reply:
@edwarnicke +1 for the purpose of 'retry' destroy.